packages/apollo-shared/src/ChangeManager/AddAssemblyFromFileChange.ts

```diff
@@ -0,0 +1,176 @@
+        FILE_UPLOAD_FOLDER,
+        `${fileChecksum}.gz`,
+      )
+
+      // Check and add new assembly
+      const assemblyDoc = await assemblyModel
+        .findOne({ name: assemblyName })
+        .exec()
+      if (assemblyDoc) {
+        throw new Error(`Assembly "${assemblyName}" already exists`)
+      }
+      // Add assembly
+      const newAssemblyDoc = await assemblyModel.create({ name: assemblyName })
+      this.logger.debug?.(
+        `Added new assembly "${assemblyName}", docId "${newAssemblyDoc._id}"`,
+      )
+
+      // Read data from compressed file and parse the content
+      await createReadStream(compressedFullFileName)
+        .pipe(createGunzip())
+        .pipe(
+          gff.parseStream({
+            parseFeatures: false,
+            parseDirectives: false,
+            parseComments: false,
+            parseSequences: true,
+          }),
+        )
+        .on('data', async (data) => {
+          if (data.sequence) {
+            this.logger.debug?.(
+              `RefSeq: "${data.id}", length: ${data.sequence.length}`,
+            )
+            this.logger.debug?.(`RefSeq: "${data.id}"`)
+            const refSeqDoc = await refSeqModel
+              .findOne({ assembly: newAssemblyDoc._id, name: data.id })
+              .exec()
+            if (refSeqDoc) {
+              throw new Error(
+                `RefSeq "${data.id}" already exists in assemblyId "${newAssemblyDoc._id}"`,
+              )
+            }
+            // Add refSeq
+            const newRefSeqDoc = await refSeqModel.create({
+              name: data.id,
+              description: data.id,
+              assembly: newAssemblyDoc._id,
+              length: data.sequence.length,
+            })
+            this.logger.debug?.(
+              `Added new refSeq "${data.id}", docId "${newRefSeqDoc._id}"`,
+            )
+
+            let ind = 0
+            const temp1 = `/.{1,${CHUNK_LEN}}/g`
+            this.logger.debug?.(`temp "${temp1}"`)
+            //   const chunkArray = await data.sequence.match(/.{1,5000}/g) // *** THIS WORKS FINE WHEN THERE IS NO SPACE BETWEEN 1 AND 5000. i.e. {1,5000} works but {1, 5000} doesn't ***** //
+            const chunkArray = await data.sequence.match(/.{1,5000}/g) // *** THIS WORKS FINE WHEN THERE IS NO SPACE BETWEEN 1 AND 5000. i.e. {1,5000} works but {1, 5000} doesn't ***** //
+            // const chunkArray = await data.sequence.match(`${temp1}`) // *** HOW TO USE VARIABLE 'CHUNK_LEN' instead of 5000 ??? *** //
+            for (const chunk of chunkArray) {
+              // Add refSeqChunks
+              const newRefSeqChunkDoc = await refSeqChunkModel.create({
+                refSeq: newRefSeqDoc._id,
+                n: ind,
+                sequence: chunk,
+              })
+              this.logger.debug?.(
+                `Added new refSeqChunk (n=${ind}) docId "${newRefSeqChunkDoc._id}" for refSeq "${newRefSeqDoc._id}"`,
+              )
+              ind++
```

Comment:
> I was not able to use CHUNK_LEN environment variable in match() -method.

I'd probably do something like this instead of using `match()`.

```suggestion
            const chunkLength = Number(CHUNK_LEN)
            const numChunks = Math.ceil(data.sequence.length / chunkLength)
            for (let chunkNum = 0; chunkNum < numChunks - 1; chunkNum++) {
              const start = chunkNum * chunkLength
              const chunk = data.sequence.slice(start, start + chunkLength)
              await refSeqChunkModel.create({
                refSeq: newRefSeqDoc._id,
                n: chunkNum,
                sequence: chunk,
              })
            }
```